### DISCUSSION TITLE
Request For Assistance

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#4110] [General]
Mackinnon Buck from the Blazor Product Team is investigating the rendering issue in Oqtane 5.1 which currently requires StreamRendering to be included. The discussion related to this item is in this thread:

https://github.com/dotnet/aspnetcore/issues/54157

As per its usual policy, Microsoft requests a "minimal repro" of an issue in order to investigate (as they do not want to have to deal with setting up app/environment dependencies)... so I was able to repro the rendering issue in the Oqtane SSR POC project. However when  Mackinnon clones the repo and runs the project he says that he cannot reproduce the problem.

Are any of you able to clone the repo (https://github.com/oqtane/OqtaneSSR) and run the app and let me know if the Home page displays the following content for you:

![319755648-69ad3752-089a-4d79-82ec-5c679ca34f5a](https://github.com/oqtane/oqtane.framework/assets/4840590/7f8ff81e-54b8-40d2-90d9-d5e2e9c3b08c)

I need to determine if my local development environment is the exception case, or if others can reproduce the same problem.

Thank you.

**Comment by oqtanecms** [comment]
**Running the app it looks like:**
![SSR0](https://github.com/oqtane/oqtane.framework/assets/17893776/924ee8f9-1032-4dbf-967f-3405a5e3dae9)

**Uncomment the [StreamRendering] attribute in Routes.razor as shown below:**
![SSR1](https://github.com/oqtane/oqtane.framework/assets/17893776/ed1a1187-6f09-4f70-be40-07a4471de241)

**Comment by krishnavsr** [comment]
fresh clone
<img width="1058" alt="image" src="https://github.com/oqtane/oqtane.framework/assets/36802032/4a5249a2-a311-45fe-b48d-8597702147bf">

after uncomment 
<img width="217" alt="image" src="https://github.com/oqtane/oqtane.framework/assets/36802032/26b54fea-39b4-49cb-a5f4-38afcc5e478b">

<img width="1159" alt="image" src="https://github.com/oqtane/oqtane.framework/assets/36802032/244799ba-e7d8-4fd5-80fe-eafb42d63a61">

dotnet --info 

.NET SDK:
 Version:           8.0.203
 Commit:            5e1ceea679
 Workload version:  8.0.200-manifests.8cf8de6d

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.203\

.NET workloads installed:
 [maui-windows]
   Installation Source: VS 17.9.34714.143
   Manifest Version:    8.0.7/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.maui\8.0.7\WorkloadManifest.json
   Install Type:        FileBased

 [maccatalyst]
   Installation Source: VS 17.9.34714.143
   Manifest Version:    17.2.8004/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.maccatalyst\17.2.8004\WorkloadManifest.json
   Install Type:        FileBased

 [ios]
   Installation Source: VS 17.9.34714.143
   Manifest Version:    17.2.8004/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.ios\17.2.8004\WorkloadManifest.json
   Install Type:        FileBased

 [android]
   Installation Source: VS 17.9.34714.143
   Manifest Version:    34.0.52/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.android\34.0.52\WorkloadManifest.json
   Install Type:        FileBased

Host:
  Version:      8.0.3
  Architecture: x64
  Commit:       9f4b1f5d66

.NET SDKs installed:
  8.0.202 [C:\Program Files\dotnet\sdk]
  8.0.203 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 8.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Comment by mdmontesinos** [comment]
Just cloned the repo and launched the project:

![image](https://github.com/oqtane/oqtane.framework/assets/90258222/735f9432-b7af-48e8-a41f-08b7844c25fa)

After uncommenting StreamRendering:

![image](https://github.com/oqtane/oqtane.framework/assets/90258222/8b1853ff-7eb3-48bf-a8a3-6c876cdc03fe)

Tested both with Debugging and Without Debugging and in Debug and Release mode.

```
dotnet --info (simplified)

.NET SDK:
 Version:           8.0.202
 Commit:            25674bb2f4
 Workload version:  8.0.200-manifests.8cf8de6d

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.202\

Host:
  Version:      8.0.3
  Architecture: x64
  Commit:       9f4b1f5d66

.NET SDKs installed:
  8.0.202 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 8.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by sbwalker** [comment]
Thank you @sundnnstore @krishnavsr @mdmontesinos  ... the results are inconsistent... but I think I see a pattern.

If you have time could you all do one more experiment...

In your Oqtane installation (latest dev branch code), please comment out the StreamRendering attribute in Oqtane.Client\UI\Routes.razor

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/bf794617-2c58-4a17-8177-26cbf7e0cb7b)

and run the framework to see what result is produced. In my local environment I get:

![320080102-c16c3fd7-3a99-4a2e-9552-9ff0eea33a7f](https://github.com/oqtane/oqtane.framework/assets/4840590/858b29c9-969e-4d64-8006-19b0c68cec59)

**Reply by krishnavsr** [reply]
@sbwalker 
It continues to work fine for me with that commented out
<img width="229" alt="image" src="https://github.com/oqtane/oqtane.framework/assets/36802032/3578c5f2-79ad-441b-9fe6-244f3000fd39">


<img width="1059" alt="image" src="https://github.com/oqtane/oqtane.framework/assets/36802032/7cd744ad-75bb-46ac-9731-a063dda1262b">

**Comment by sbwalker** [comment]
More strange behavior... I just updated my local environment to VS 17.8.7 and reinstalled the .NET SDK 8.0.3... and now the Oqtane Framework will not compile:

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CS9057	The analyzer assembly 'C:\Program Files\dotnet\sdk\8.0.203\Sdks\Microsoft.NET.Sdk.Razor\source-generators\Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll' references version '4.9.0.0' of the compiler, which is newer than the currently running version '4.8.0.0'.	Oqtane.Client	C:\Source\Projects\oqtane.framework\Oqtane.Client\CSC	1	Active
 ```

**Reply by oqtanecms** [reply]
just updated my local environment to VS 17.9.5 and the .NET SDK 8.0.2.The framework with latest dev branch code compiles without problems:
![0](https://github.com/oqtane/oqtane.framework/assets/17893776/c9667f7f-a6ef-433f-99ce-507401e56101)
It runs without any problem:
![1](https://github.com/oqtane/oqtane.framework/assets/17893776/8cf87e94-39ce-43e7-b132-6088f52a782c)
![2](https://github.com/oqtane/oqtane.framework/assets/17893776/5eba4977-8a79-486c-af02-6ecdf2bcdafa)

**Reply by thabaum** [reply]
@sbwalker dev branch can now be updated to version 5.1.1 for reporting issues.  I can build and run Oqtane in Visual Studio with the latest updates.  I have not had time to test SSR issue requested yet.

**Comment by mdmontesinos** [comment]
I just tested the lastest dev branch by commenting the StreamRendering in Routes.razor and it works perfectly fine. All the content is displayed without any problem.

**Comment by sbwalker** [comment]
Thank you everyone for your assistance. I believe the differences in behavior are related to the different SDK versions that people have installed. I was told by a source at Microsoft that Blazor in .NET 8 “came in hot” meaning that it had to ship by a certain date even if it wasn’t 100% ready. This means that the patch releases of .NET 8 (8.0.1, 8.0.2, 8.0.3) contain a lot of critical fixes to mainline use cases. Which explains the unexpected behavior we saw early in the Oqtane integration versus now. Also, when the package dependencies are updated in a .NET solution to a newer patch version it does not mean your development environment is going to benefit from them unless you also install the related SDK. In my case I was still developing on an older SDK version even though Oqtane was updated to .NET 8.0.3 packages (and I noticed that many of you are also not fully updated to the  8.0.3 SDK).

**Reply by thabaum** [reply]
@sbwalker I tested as well and commenting out the stream rendering attribute seems to allow things to work and removed the flicker from when you are logged in between page clicks making it looks like enhanced navigation is working more like expected.

User Profile seems to reload its data twice.  Looks like the stream rendering attribute can be removed with latest .NET fixes.  I also tested in the SSR project and works the same.

**Reply by thabaum** [reply]
I removed from Head.razor, Routes.razor and the 2 StreamRenderingDisabled.razor and StreamRenderingEnabled.razor.  I have yet to find an issue other than User Profile component loading twice and with only minimal testing.

**Reply by sbwalker** [reply]
@thabaum did you upgrade to a newer SDK version to get these results? The other item I am investigating is if the SDK version which the app is built with affects the runtime behavior. For example, my Azure environment says it is running 8.0.100 - but in my Azure config the only option I have is to set the major .NET version to 8.0.

**Reply by thabaum** [reply]
I installed latest visual studio, I can test production tomorrow, out for the day...

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/4cd52ac6-47cb-4f20-8942-b4e777f42938)

```
Microsoft Visual Studio Community 2022
Version 17.9.5
VisualStudio.17.Release/17.9.5+34723.18
Microsoft .NET Framework
Version 4.8.09032

Installed Version: Community

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASA Service Provider   1.0

ASP.NET and Web Tools   17.9.199.22661
ASP.NET and Web Tools

Azure App Service Tools v3.0.0   17.9.199.22661
Azure App Service Tools v3.0.0

Azure Data Lake Tools for Visual Studio   2.6.5000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   17.9.199.22661
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.6.5000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   4.9.0-3.24121.1+a98c90d56455379836dd5c845b35fa932b00cfa3
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Cookiecutter   17.0.24002.3
Provides tools for finding, instantiating and customizing templates in cookiecutter format.

Extensibility Message Bus   1.4.39 (main@e8108eb)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

GitHub Copilot   1.147.0.0 (v1.147.0.0@218a5ecac)
GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.

GitHub Copilot Agent   1.147.0

Microsoft Azure Hive Query Language Service   2.6.5000.0
Language service for Hive query

Microsoft Azure Stream Analytics Language Service   2.6.5000.0
Language service for Azure Stream Analytics

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Mono Debugging for Visual Studio   17.9.0 (b3bca6f)
Support for debugging Mono processes with Visual Studio.

Node.js Tools   1.5.50908.1 Commit Hash:c294679b821c4b8f7f9c6db3827b7655db80fc28
Adds support for developing and debugging Node.js apps in Visual Studio

NuGet Package Manager   6.9.2
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Python - Profiling support   17.0.24002.3
Profiling support for Python projects.

Python with Pylance   17.0.24002.3
Provides IntelliSense, projects, templates, debugging, interactive windows, and other support for Python developers.

Razor (ASP.NET Core)   17.9.2.2412004+684740676513b3b208d5bd5b7f8786dab8d287b7
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.9.55.1
Microsoft SQL Server Data Tools

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   17.0.30103.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.9.0-3.24121.1+a98c90d56455379836dd5c845b35fa932b00cfa3
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.9.0-beta.23614.3+a521e1cd420beb56c15faf6836184fadd2b7937a
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VSPackage Extension   1.0
VSPackage Visual Studio Extension Detailed Info

Xamarin   17.9.0.138 (d17-9@60c2b9b)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.9.3.0 (remotes/origin/d17-9@f174974f91)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin.Android SDK   13.2.2.0 (d17-5/45b0e14)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: d9a6e87
    Java.Interop: xamarin/java.interop/d17-5@149d70fe
    SQLite: xamarin/sqlite/3.40.1@68c69d8
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d17-5@ca1552d

```

**Reply by thabaum** [reply]
@sbwalker I install the latest hosting bundle.

https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-aspnetcore-8.0.3-windows-hosting-bundle-installer

> 
ASP.NET Core Hosting Bundle contains everything you need to run existing web/ server apps. The bundle includes the .NET runtime, the ASP.NET Core runtime, and if installed on a machine with IIS it will also add the [ASP.NET Core IIS Module](https://learn.microsoft.com/aspnet/core/fundamentals/servers/aspnet-core-module?WT.mc_id=dotnet-35129-website). If you only want the .NET or ASP.NET Core runtime, you'll find them in [.NET 8.0 Downloads](https://dotnet.microsoft.com/en-us/download/dotnet/8.0).

**Reply by sbwalker** [reply]
I just did a build from the Dev branch and deployed it to Azure and it runs fine. When I use Kudu to check which version of the SDK it is running it says:

C:\home>dotnet --info
.NET SDK:
 Version:           8.0.101
 Commit:            6eceda187b
 Workload version:  8.0.100-manifests.69afb982

So it appears that Azure is a couple of SDK patch releases behind - and I am not aware of any way to update it in an App Service. Regardless, Oqtane is working fine on 8.0.1 so its not really an issue.

**Reply by thabaum** [reply]
@sbwalker maybe you can set it manually in azure with command line?
https://learn.microsoft.com/en-us/azure/app-service/configure-language-dotnetcore?pivots=platform-windows#set-net-core-version


```
dotnet --info
.NET SDK:
 Version:           8.0.203
 Commit:            5e1ceea679
 Workload version:  8.0.200-manifests.a7f084b6

Host:
  Version:      8.0.3
  Architecture: x64
  Commit:       9f4b1f5d66
```

**Reply by sbwalker** [reply]
its work on 8.0.1 so its not really an issue

**Reply by thabaum** [reply]
After todays update:

```
 dotnet --info
.NET SDK:
   Version:           8.0.204
   Commit:            c338c7548c
   Workload version:  8.0.200-manifests.8f02a06a

Host:
  Version:      8.0.4
  Architecture: x64
  Commit:       2d7eea2529

```

Just curious if you need to restart the app service or do something in Azure to update to latest.  I do not currently have an account to test.  I also wonder if you moved it's location if it would use the latest since it may move things to a server with the newer version?  Then maybe move it back to it's original location to see if can get on a server running the latest version at that hosting location... just thinking out loud a little.

Seems like Azure should always be the latest.
